libkbfs/backpressure_disk_limiter.go

```diff
@@ -470,6 +479,49 @@ func (bdl *backpressureDiskLimiter) onBlocksDelete(
 	bdl.fileTracker.onBlocksDelete(blockFiles)
 }
 
+func (bdl *backpressureDiskLimiter) onDiskBlockCacheDelete(
+	ctx context.Context, blockBytes int64) {
+	if blockBytes == 0 {
+		return
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	bdl.byteTracker.onBlocksDelete(blockBytes)
+}
+
+func (bdl *backpressureDiskLimiter) beforeDiskBlockCachePut(
+	ctx context.Context, blockBytes, diskBlockCacheBytes int64) (
+	availableBytes int64, err error) {
+	if blockBytes == 0 {
+		// Better to return an error than to panic in Acquire.
+		return 0, errors.New("backpressureDiskLimiter.beforeDiskBlockCachePut" +
+			" called with 0 blockBytes")
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	freeBytes, freeFiles, err := bdl.freeBytesAndFilesFn()
+	if err != nil {
+		return
+	}
+	bdl.byteTracker.updateFree(freeBytes)
+	bdl.fileTracker.updateFree(freeFiles)
+
+	diskBlockCacheLimit := int64(bdl.byteTracker.currLimit() *
+		bdl.byteTracker.minThreshold * diskBlockCacheFrac)
+	if diskBlockCacheLimit < diskBlockCacheBytes+blockBytes {
+		// The overall disk block cache limit has been exceeded.  Return a
+		// negative result so the block cache knows how much it must evict
+		// before retrying.
+		return diskBlockCacheLimit - (diskBlockCacheBytes + blockBytes),
+			errors.New("backpressureDiskLimiter needs more space")
+	}
+	// The disk block cache is using up less than diskBlockCacheLimit bytes, so
```

Comment:
Hmm...I think we can avoid the possibility of making backpressure worse by having the semaphore have a `TryAcquire` function. Add a TODO perhaps, since it's not urgent. I had thought that having the diskBlockCacheLimit be some fraction of currLimit() was enough to guarantee that the semaphore couldn't go negative, but I'd have to think more about that...